Update common files.

geeknoid opened the pull request:
- Switch build to use the protos delivered through common files when possible.

sdake commented on Makefile.core.mk:
```diff
@@ -159,178 +141,162 @@ census_packages = \
-$(googleprotobuf_protos): %:
-	@mkdir -p googleprotobuf/google/protobuf
-	@curl -sS $(GOOGLEPROTOBUF_URL)/$@ -o googleprotobuf/$@
+TMPDIR=/tmp/protos
```
should really use mktemp -d here.

sdake commented on Makefile.core.mk:
```diff
@@ -159,178 +141,162 @@ census_packages = \
-	@sed -i -e '/^option go_package/d' k8sapi/k8s.io/api/$@
+prometheus_prep:
+	@mkdir -p ${TMPDIR}/prometheus prometheus
+	@curl -sS $(PROMETHEUS_URL)/metrics.proto -o ${TMPDIR}/prometheus/metrics.proto
```
this needs to go in commonn protos imo.  I will submit a separate pr for that.

sdake commented on Makefile.core.mk:
```diff
@@ -159,178 +141,162 @@ census_packages = \
-	@sed -i -e '/^option go_package/d' oc/$@
+$(census_protos): %: google_prep
+	@mkdir -p ${TMPDIR}/opencensus/proto/stats/v1 ${TMPDIR}/opencensus/proto/trace/v1 ${TMPDIR}/opencensus/proto/resource/v1 opencensus
+	@curl -sS $(CENSUS_URL)/$@ -o ${TMPDIR}/$@
```
this should go in common-protos.  Will submit a PR for this.

sdake commented:
https://github.com/istio/common-files/pull/48 contains the correct protos so this repo need not curl them.

Cheers
-steve

sdake commented on Makefile.core.mk:
```diff
@@ -159,178 +135,167 @@ census_packages = \
-	@curl -sS $(GOOGLEPROTOBUF_URL)/$@ -o googleprotobuf/$@
-
-$(googleprotobuf_packages): %: $(googleprotobuf_protos)
+TMPDIR=${mkdir -d}
```
@geeknoid I may have mentioned `mkdir -d`.  what I had meant was:
```
TMPDIR="$(mktemp -d /tmp/gogobuild-XXXXXXXX)"
```
This will create a secure directory in /tmp with gogobuild- and 8 random charracter suffi.

geeknoid replied:
Why is that necessary? What makes it "secure"? And doesn't mktemp already create uniquely named directories?

sdake replied:
no.  without the `-d` flag, no directory is created.  I suppose the reet doesn't matter.  The directory creation is secure in execution.  A special system call is used to make the file or directory which does not allow someone to symlink attack the directory creation.  I know it seems silly to address security ini the non-runtime portion of the code, however, when code goes through a third-party audit such as those required by the CNCF, this would have been caught, causing rework and a re-release to meet the security audit.

```
Stevens-MacBook-Pro:~ sdake$ mktemp
/var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK
Stevens-MacBook-Pro:~ sdake$ ls -l /var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK 
-rw-------  1 sdake  staff  0 Sep  4 05:51 /var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK
```

jyotimahapatra commented:
I was looking to update the go-control-plane dependency to the latest version from master. Do you know what version can be used here? https://github.com/envoyproxy/go-control-plane/blob/master/go.mod#L13

kyessenov commented:
@geeknoid @rshriram Can we revert to using the "official" protos from gogo/googleapis? What's the reason to duplicate the protos here? We've added a strange dependency from envoy on istio/gogo-genproto that I really don't like.

geeknoid commented:
As we discussed extensively on Slack, we updated to the version of gogo
we're using in order to avail ourselves of a bug fix we needed. Now that
gogo 1.3 is available, I plan on upgrading everything on our end to that in
a week or two.

On Thu, Sep 5, 2019 at 4:52 PM Kuat <notifications@github.com> wrote:

> @geeknoid <https://github.com/geeknoid> @rshriram
> <https://github.com/rshriram> Can we revert to using the "official"
> protos from gogo/googleapis? What's the reason to duplicate the protos
> here? We've added a strange dependency from envoy on istio/gogo-genproto
> that I really don't like.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/gogo-genproto/pull/22?email_source=notifications&email_token=AFNZYHLPKO2HT6O7RCUOQNLQIGLSDA5CNFSM4ISQRHDKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6BJBMY#issuecomment-528650419>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFNZYHM3LGGUESQP2TVMFN3QIGLSDANCNFSM4ISQRHDA>
> .
>
